src/include/mpir_hw_topo.h

```diff
@@ -0,0 +1,189 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef MPIR_HW_TOPO_H_INCLUDED
+#define MPIR_HW_TOPO_H_INCLUDED
+
+/*
+ * Tracks hwloc_obj_type_t do not change ordering
+ */
+typedef enum {
+    MPIR_HWLOC_OBJ_MACHINE,
+    MPIR_HWLOC_OBJ_PACKAGE,
+    MPIR_HWLOC_OBJ_CORE,
+    MPIR_HWLOC_OBJ_PU,
+    MPIR_HWLOC_OBJ_L1CACHE,
+    MPIR_HWLOC_OBJ_L2CACHE,
+    MPIR_HWLOC_OBJ_L3CACHE,
+    MPIR_HWLOC_OBJ_L4CACHE,
+    MPIR_HWLOC_OBJ_L5CACHE,
+    MPIR_HWLOC_OBJ_L1ICACHE,
+    MPIR_HWLOC_OBJ_L2ICACHE,
+    MPIR_HWLOC_OBJ_L3ICACHE,
+    MPIR_HWLOC_OBJ_GROUP,
+    MPIR_HWLOC_OBJ_NUMANODE,
+    MPIR_HWLOC_OBJ_BRIDGE,
+    MPIR_HWLOC_OBJ_PCI_DEVICE,
+    MPIR_HWLOC_OBJ_OS_DEVICE,
+    MPIR_HWLOC_OBJ_MISC,
+    MPIR_HWLOC_OBJ_TYPE_MAX,
+    MPIR_MPICH_OBJ_DDR,         /* MPICH object for DDR memory */
+    MPIR_MPICH_OBJ_MCDRAM,      /* MPICH object for MCDRAM memory */
+    MPIR_MPICH_OBJ_DEFAULT = MPIR_MPICH_OBJ_DDR
+} MPIR_Hwloc_obj_type;
+
+/*
+ * Tracks hwloc_membind_policy_t do not change
+ */
+typedef enum {
+    MPIR_HWLOC_MEMBIND_DEFAULT = 0,
+    MPIR_HWLOC_MEMBIND_FIRSTTOUCH = 1,
+    MPIR_HWLOC_MEMBIND_BIND = 2,
+    MPIR_HWLOC_MEMBIND_INTERLEAVE = 3,
+    MPIR_HWLOC_MEMBIND_NEXTTOUCH = 4,
+    MPIR_HWLOC_MEMBIND_MIXED = -1
+} MPIR_Hwloc_membind_policy;
+
+/*
+ * Tracks hwloc_membind_flags_t do not change
+ */
+typedef enum {
+    MPIR_HWLOC_MEMBIND_PROCESS = (1 << 0),
+    MPIR_HWLOC_MEMBIND_THREAD = (1 << 1),
+    MPIR_HWLOC_MEMBIND_STRICT = (1 << 2),
+    MPIR_HWLOC_MEMBIND_MIGRATE = (1 << 3),
+    MPIR_HWLOC_MEMBIND_NOCPUBIND = (1 << 4),
+    MPIR_HWLOC_MEMBIND_BYNODESET = (1 << 5)
+} MPIR_Hwloc_membind_flags;
+
+/*
+ * MPICH specific definitions
+ */
+typedef enum {
+    MPIR_MPICH_NETLOC_OBJ_NETWORK,
+    MPIR_MPICH_NETLOC_OBJ_TOPOLOGY,
+    MPIR_MPICH_NETLOC_OBJ_NODE
+} MPIR_Netloc_obj_class;
+
+/*
+ * Tracks netloc_network_type_t do not change
+ */
+typedef enum {
+    MPIR_NETLOC_NETWORK_TYPE_ETHERNET = 1,
+    MPIR_NETLOC_NETWORK_TYPE_INFINIBAND = 2,
+    MPIR_NETLOC_NETWORK_TYPE_INVALID = 3
+} MPIR_Netloc_network_type;
+
+/*
+ * Tracks netloc_topology_type_t do not change
+ */
+typedef enum {
+    MPIR_NETLOC_TOPOLOGY_TYPE_INVALID = -1,
+    MPIR_NETLOC_TOPOLOGY_TYPE_TREE = 1
+} MPIR_Netloc_topology_type;
+
+/*
+ * Tracks netloc_node_type_t do not change
+ */
+typedef enum {
+    MPIR_NETLOC_NODE_TYPE_HOST = 0,
+    MPIR_NETLOC_NODE_TYPE_SWITCH = 1,
+    MPIR_NETLOC_NODE_TYPE_INVALID = 2
+} MPIR_Netloc_node_type;
+
+/*
+ * MPICH specific definitions
+ */
+typedef enum {
+    MPIR_MPICH_NETLOC__SUBCOMM_MIN_SIZE,
+    MPIR_MPICH_NETLOC__MIN_MEM_SIZE,
+    MPIR_MPICH_NETLOC__DIMENSION
+} MPIR_Netloc_search_key;
```

Comment:
These are not specific to netloc or the network.  They are just logical partitioning models.  How about `MPIR_COMM_SPLIT_MODEL__*`?